helm/hpcc/templates/_helpers.tpl

```diff
@@ -615,3 +615,44 @@ resources:
     memory: "100M"
 {{- end -}}
 
+
+{{/*
+Create placement related settings
+Pass in dict with placement
+*/}}
+{{- define "hpcc.doPlacement" -}}
+{{- if .placement.nodeSelector }}
+nodeSelector:
+{{ toYaml .placement.nodeSelector | indent 2 }}
+{{- end -}}
+{{- if .placement.tolerations }}
+tolerations:
+{{ toYaml .placement.tolerations }}
+{{- end -}}
+{{- if .placement.affinity }}
+affinity:
+{{ toYaml .placement.affinity }}
+{{- end -}}
+{{- if .placement.schedulerNames -}}
+{{- range $profileName := .placement.schedulerNames }}
+schedulerName: {{ $profileName }}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Check if there is any placement configuration
+Pass in dict with root, pod, cluster and type
+*/}}
+{{- define "hpcc.placementMatchName" -}}
+{{- if .root.Values.placement }}
+{{- $pod := .pod -}}
+{{- $cluster := .cluster -}}
+{{- $type := .type -}}
+{{- range $placement := .root.Values.placement -}}
+{{- if or (has $pod $placement.pods) (has $cluster $placement.pods) (has $type $placement.pods) -}}
```

Comment:
I think this may need to be more explicit to separate type, target and name for matching.
If I want to place all eclagent types, and they are defined as "hthor" and "somethingelse", "anotherone" then I don't think there's a way to match them all, without listing them all.

If instead you could declare a placement like e.g.:
```
- type: ["eclagent"] # all pods that match type="eclagent"
  nodeSelector:
    name: nptwo
```

then it would be concise and avoid overmatching.

Reply:
Yes, current implement cannot prevent duplicate settings. It is up to user to provide single copy for each setting.